models/Shift.js

```diff
@@ -17,12 +17,23 @@ const ShiftSchema= new mongoose.Schema({
   },
   wage: Number,
   tips: Boolean,
-  date: Date,
-  start_time: String,
-  end_time: String,
-  activeStatus: { type: Boolean, default: true }, 
+  start_at: Date,
+  end_at: Date,
+  duration: Number,
+  activeStatus: { type: Boolean, default: true },
   more: String,
   availability: [ { type: String } ]
 });
 
+ShiftSchema.pre('save', function save(next) {
+  const shift = this
+  const oneHour = 1000 * 60 * 60
+  this.duration = secDecimal((new Date(shift.end_at) - new Date(shift.start_at)) / oneHour )

```

Comment:
I'd also use a library for this. Date manipulation is a well-solved problem in JS, no reason to reinvent the wheel.